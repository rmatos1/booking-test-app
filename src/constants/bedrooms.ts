import { images } from '../assets';

export interface IBedroom {
  number: number;
  type: string;
  beds: {
    double?: number;
    single?: number;
  };
  img: string;
  fullCapacity: number;
  dailyPrice: number;
}

/**
 * constant that stores the maximum number of people possible in the biggest bedroom
 */
export const MAX_CAPACITY_BEDROOM = 4;

/**
 * constant with the available bedrooms to be booked
 */
export const hotelBedrooms: IBedroom[] = [
  {
    number: 1,
    type: 'Quadruple',
    beds: {
      double: 1,
      single: 2,
    },
    img: images.bedroom1,
    fullCapacity: 4,
    dailyPrice: 65,
  },
  {
    number: 2,
    type: 'Single',
    beds: {
      double: 1,
    },
    img: images.bedroom2,
    fullCapacity: 2,
    dailyPrice: 40,
  },
  {
    number: 3,
    type: 'Triple',
    beds: {
      double: 1,
      single: 1,
    },
    img: images.bedroom3,
    fullCapacity: 3,
    dailyPrice: 55,
  },
  {
    number: 4,
    type: 'Triple',
    beds: {
      single: 3,
    },
    img: images.bedroom4,
    fullCapacity: 3,
    dailyPrice: 50,
  },
  {
    number: 5,
    type: 'Single',
    beds: {
      single: 2,
    },
    img: images.bedroom5,
    fullCapacity: 2,
    dailyPrice: 35,
  },
];
