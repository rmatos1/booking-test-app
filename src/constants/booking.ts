export const INITIAL_BOOKING_DATA = {
  checkIn: '',
  checkOut: '',
  qtyGuests: '',
  name: '',
  email: '',
  selectedBedroom: 0,
};

export const TEST_BOOKING_DATA = {
  id: '',
  name: 'Julia',
  email: 'julia@teste.com',
  checkIn: '2023-12-19',
  checkOut: '2023-12-26',
  selectedBedroom: 3,
  qtyGuests: 3,
  totalPrice: 450,
};
