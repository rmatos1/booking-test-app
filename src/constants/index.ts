export * from './bedrooms';
export * from './booking';
export * from './styles';
