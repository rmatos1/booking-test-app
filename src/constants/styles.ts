/*
 * break points
 */
export const SMALL_DESKTOP = '990px';
export const TABLET = '768px';
export const MOBILE = '460px';

/***************************** */

export const HORIZONTAL_PADDING_DESKTOP = '25px';

export const HORIZONTAL_PADDING_MOBILE = '15px';
