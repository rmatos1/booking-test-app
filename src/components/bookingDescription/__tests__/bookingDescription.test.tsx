import { describe, test } from '@jest/globals';
import { RenderResult, render } from '@testing-library/react';
import moment from 'moment';

import { BookingDescription, IBookingDescription } from '..';
import { testBookingData } from '../../../constants';

const defaultProps: IBookingDescription = {
  data: {
    ...testBookingData
  },
};

const setup = (componentProps?: IBookingDescription): JSX.Element => {
  const baseProps = componentProps || defaultProps;

  return <BookingDescription {...baseProps} />;
};

describe('<BookingDescription />', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = render(setup());
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the defined title', () => {
    const TITLE = 'Description';

    wrapper = render(setup({ ...defaultProps, title: TITLE }));

    const title = wrapper.getByText(TITLE);

    expect(title).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the defined name', () => {
    const name = wrapper.getByTestId('name');

    expect(name.textContent).toContain(defaultProps.data.name);
  });

  test('should render the defined email', () => {
    const email = wrapper.getByTestId('email');

    expect(email.textContent).toContain(defaultProps.data.email);
  });

  test('should render the defined check-in and check-out', () => {
    const period = wrapper.getByTestId('period');

    expect(period.textContent).toContain(
      moment(defaultProps.data.checkIn).format('MM/DD/YYYY')
    );
    expect(period.textContent).toContain(
      moment(defaultProps.data.checkOut).format('MM/DD/YYYY')
    );
  });

  test('should render the number of the defined selected bedroom', () => {
    const bedroomNumber = wrapper.getByTestId('bedroom-number');

    expect(bedroomNumber.textContent).toContain(
      defaultProps.data.selectedBedroom.toString()
    );
  });

  test('should render the defined quantity of guests', () => {
    const qtyGuests = wrapper.getByTestId('qty-guests');

    expect(qtyGuests.textContent).toContain(
      defaultProps.data.qtyGuests.toString()
    );
  });

  test('should render the defined total price', () => {
    const totalPrice = wrapper.getByTestId('total-price');

    expect(totalPrice.textContent).toContain(
      defaultProps.data.totalPrice.toString()
    );
  });
});
