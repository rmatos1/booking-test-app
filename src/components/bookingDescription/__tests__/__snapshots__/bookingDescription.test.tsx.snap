// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingDescription /> should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="sc-bdfCDU jqNSQk"
      >
        <tbody>
          <tr>
            <td
              class="sc-dlfmHC iQFkgf"
            >
              Name:
            </td>
            <td
              class="sc-hKgKIp hjxdtj"
              data-testid="name"
            >
              Julia
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfmHC iQFkgf"
            >
              Email:
            </td>
            <td
              class="sc-hKgKIp hjxdtj"
              data-testid="email"
            >
              julia@teste.com
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfmHC iQFkgf"
            >
              Period:
            </td>
            <td
              class="sc-hKgKIp hjxdtj"
              data-testid="period"
            >
              12/19/2023
               -
               
              12/26/2023
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfmHC iQFkgf"
            >
              Qty of Nights:
            </td>
            <td
              class="sc-hKgKIp hjxdtj"
            >
              7
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfmHC iQFkgf"
            >
              Bedroom nº:
            </td>
            <td
              class="sc-hKgKIp hjxdtj"
              data-testid="bedroom-number"
            >
              3
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfmHC iQFkgf"
            >
              Qty of Guests:
            </td>
            <td
              class="sc-hKgKIp hjxdtj"
              data-testid="qty-guests"
            >
              3
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfmHC iQFkgf"
            >
              Total Price:
            </td>
            <td
              class="sc-hKgKIp hjxdtj"
              data-testid="total-price"
            >
              $ 
              450
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="sc-bdfCDU jqNSQk"
    >
      <tbody>
        <tr>
          <td
            class="sc-dlfmHC iQFkgf"
          >
            Name:
          </td>
          <td
            class="sc-hKgKIp hjxdtj"
            data-testid="name"
          >
            Julia
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlfmHC iQFkgf"
          >
            Email:
          </td>
          <td
            class="sc-hKgKIp hjxdtj"
            data-testid="email"
          >
            julia@teste.com
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlfmHC iQFkgf"
          >
            Period:
          </td>
          <td
            class="sc-hKgKIp hjxdtj"
            data-testid="period"
          >
            12/19/2023
             -
             
            12/26/2023
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlfmHC iQFkgf"
          >
            Qty of Nights:
          </td>
          <td
            class="sc-hKgKIp hjxdtj"
          >
            7
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlfmHC iQFkgf"
          >
            Bedroom nº:
          </td>
          <td
            class="sc-hKgKIp hjxdtj"
            data-testid="bedroom-number"
          >
            3
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlfmHC iQFkgf"
          >
            Qty of Guests:
          </td>
          <td
            class="sc-hKgKIp hjxdtj"
            data-testid="qty-guests"
          >
            3
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlfmHC iQFkgf"
          >
            Total Price:
          </td>
          <td
            class="sc-hKgKIp hjxdtj"
            data-testid="total-price"
          >
            $ 
            450
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
